src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/CreateFileFromTemplateService.cs

```diff
@@ -34,13 +34,13 @@ public CreateFileFromTemplateService(IUnconfiguredProjectVsServices projectVsSer
         /// Create a file with the given template file and add it to the parent node.
         /// </summary>
         /// <param name="templateFile">The name of the template zip file.</param>
-        /// <param name="parentNode">The node to which the new file will be added.</param>
+        /// <param name="parentDocumentMoniker">The path to the node to which the new file will be added.</param>
         /// <param name="fileName">The name of the file to be created.</param>
         /// <returns>true if file is added successfully.</returns>
-        public async Task<bool> CreateFileAsync(string templateFile, IProjectTree parentNode, string fileName)
+        public async Task<bool> CreateFileAsync(string templateFile, string parentDocumentMoniker, string fileName)
         {
             Requires.NotNull(templateFile, nameof(templateFile));
-            Requires.NotNull(parentNode, nameof(parentNode));
+            Requires.NotNullOrEmpty(parentDocumentMoniker, nameof(parentDocumentMoniker));
```

Comment:
Here's a little sample that shows you what I mean:

``` C#
    class Program
    {
        static void Main(string[] args)
        {
            for (char c = (char)0; c < Char.MaxValue; c++)
            {
                if (char.IsWhiteSpace(c) && CanCreateFile(c))
                {
                    Console.WriteLine("Valid Path Char: " + ToCodePoint(c));
                }
            }
        }

        private static string ToCodePoint(char c)
        {
            return string.Format("\\u{0:x4} ", (int)c);
        }

        private static bool CanCreateFile(char c)
        {
            try
            {
                string path = Path.Combine(@"c:\Temp", c.ToString());

                File.WriteAllText(path, "");
            }
            catch (DirectoryNotFoundException)
            {   // Handle ' ' and '\t', etc 
                return false;
            }
            catch (ArgumentException)
            {
                return false;
            }

            return true;
        }
    }
```

This outputs the following:

```
Valid Path Char: \u1680
Valid Path Char: \u2000
Valid Path Char: \u2001
Valid Path Char: \u2002
Valid Path Char: \u2003
Valid Path Char: \u2004
Valid Path Char: \u2005
Valid Path Char: \u2006
Valid Path Char: \u2007
Valid Path Char: \u2008
Valid Path Char: \u2009
Valid Path Char: \u200a
Valid Path Char: \u2028
Valid Path Char: \u2029
Valid Path Char: \u202f
Valid Path Char: \u205f
Valid Path Char: \u3000
```

These "whitespace" chars are all valid file name chars on my NTFS drive, yet IsNullOrWhiteSpace would rule them out.